pkg/imagedetails/imagedetails.go

```diff
@@ -10,18 +10,27 @@ import (
 	"github.com/acorn-io/acorn/pkg/images"
 	"github.com/acorn-io/acorn/pkg/tags"
 	"github.com/acorn-io/baaah/pkg/router"
+	imagename "github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sirupsen/logrus"
 	apierror "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const defaultNoReg = "xxx-no-reg"
+
 func GetImageDetails(ctx context.Context, c kclient.Client, namespace, imageName string, profiles []string, deployArgs map[string]any, opts ...remote.Option) (*apiv1.ImageDetails, error) {
 	imageName = strings.ReplaceAll(imageName, "+", "/")
 	name := strings.ReplaceAll(imageName, "/", "+")
 
 	if tagPattern, isPattern := autoupgrade.AutoUpgradePattern(imageName); isPattern {
-		if latestImage, found, err := autoupgrade.FindLatestTagForImageWithPattern(ctx, c, namespace, imageName, tagPattern); err != nil {
+		current, err := imagename.ParseReference(imageName, imagename.WithDefaultRegistry(defaultNoReg))
```

Comment:
I'm not sure if this will work. For example, `v#.#.#` is a supported autoupgrade pattern, but `imagename.ParseReference` will error trying to parse an image with that tag.

Reply:
I might have to go back and check on this because when i was running the debugger i think i did pass successfully through the debugger through this line but there is a possibility that i might be thinking of something else. Anyhow since we are removing the usage of current this line will not be present. 👍